File: kakao.app/js/ked/view/menu/greet.js
var _k_ = {extend: function (c,p) {for (var k in p) { if (Object.prototype.hasOwnProperty(p, k)) c[k] = p[k] } function ctor() { this.constructor = c; } ctor.prototype = p.prototype; c.prototype = new ctor(); c.__super__ = p.prototype; return c;}, list: function (l) {return l != null ? typeof l.length === 'number' ? l : [] : []}}

var greet

import kxk from "../../../kxk.js"
let kstr = kxk.kstr
let post = kxk.post

import help from "../../util/help.js"

import theme from "../../theme/theme.js"

import belt from "../../edit/tool/belt.js"

import editor from "../../edit/editor.js"

import view from "../base/view.js"


greet = (function ()
{
    _k_.extend(greet, view)
    function greet (screen)
    {
        this["draw"] = this["draw"].bind(this)
        this["show"] = this["show"].bind(this)
        greet.__super__.constructor.call(this,screen,'greet')
        this.header = help.headerCells()
        this.name = 'greet'
        this.a = 120
        this.cells.rows = 0
    }

    greet.prototype["hide"] = function ()
    {
        post.emit('greet.hide')
        return greet.__super__.hide.call(this)
    }

    greet.prototype["show"] = function ()
    {
        post.emit('greet.show')
        return greet.__super__.show.call(this)
    }

    greet.prototype["layout"] = function (x, y)
    {
        var h, w

        var _a_ = belt.cellSize(this.header); w = _a_[0]; h = _a_[1]

        return this.cells.layout(x,y,w,h)
    }

    greet.prototype["draw"] = function ()
    {
        var cell, duration, f, row, x, y

        if (this.hidden())
        {
            return
        }
        duration = 480
        this.a += 1
        if (this.a > duration)
        {
            this.a = 0
        }
        f = 0.4 + 0.6 * Math.abs(Math.sin(2 * Math.PI * this.a / duration))
        this.header = help.headerCells(f)
        var list = _k_.list(this.header)
        for (y = 0; y < list.length; y++)
        {
            row = list[y]
            var list1 = _k_.list(row)
            for (x = 0; x < list1.length; x++)
            {
                cell = list1[x]
                this.cells.set(x,y,cell.char,cell.fg,cell.bg)
            }
        }
        return post.emit('redraw')
    }

    return greet
})()

export default greet;

File: kakao.app/love/lua/view/menu/greet.lua
--[[
     ███████   ████████   ████████  ████████  █████████  
    ███        ███   ███  ███       ███          ███     
    ███  ████  ███████    ███████   ███████      ███     
    ███   ███  ███   ███  ███       ███          ███     
     ███████   ███   ███  ████████  ████████     ███     

    the fancy glowing app name ascii art thingy
--]]

help = require "util.help"
belt = require "edit.tool.belt"
view = require "view.base.view"


local greet = class("greet", view)
    


function greet:init() 
        view.init(self, 'greet')
        
        self.isVisible = false
        
        self.header = help.headerCells()
        self.name = 'greet'
        self.a = 120
        
        self.cells.rows = 0
        return self
    end

--  ███████  ███   ███   ███████   ███   ███  
-- ███       ███   ███  ███   ███  ███ █ ███  
-- ███████   █████████  ███   ███  █████████  
--      ███  ███   ███  ███   ███  ███   ███  
-- ███████   ███   ███   ███████   ██     ██  


function greet:hide() 
        post:emit('greet.hide')
        return view.hide(self)
    end


function greet:show() 
        post:emit('greet.show')
        return view.show(self)
    end


function greet:layout(x, y) 
        local w, h = belt.cellSize(self.header)
        
        return self.cells:layout(x, y, w, h)
    end

-- ███████    ████████    ███████   ███   ███  
-- ███   ███  ███   ███  ███   ███  ███ █ ███  
-- ███   ███  ███████    █████████  █████████  
-- ███   ███  ███   ███  ███   ███  ███   ███  
-- ███████    ███   ███  ███   ███  ██     ██  


function greet:draw() 
        if self:hidden() then return end
        
        local duration = 480
        self.a = self.a + 1
        if (self.a > duration) then self.a = 0 end
        
        local f = (0.4 + (0.6 * math.abs(math.sin((((2 * math.pi) * self.a) / duration)))))
        
        self.header = help.headerCells(f)
        
        for y, row in ipairs(self.header) do 
            for x, cell in ipairs(row) do 
                self.cells:set(x, y, cell.char, cell.fg, cell.bg)
            end
        end
        
        return self:render()
    end

return greet